specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json

```diff
@@ -0,0 +1,511 @@
+          "items": {
+            "$ref": "#/definitions/PlaybookConfigurationSource"
+          }
+        },
+        "actions": {
+          "type": "array",
+          "description": "A collection of the actions which are triggered if all the configured rule set evaluation is true.",
+          "items": {
+            "$ref": "#/definitions/PlaybookConfigurationAction"
+          }
+        }
+      }
+    },
+    "PlaybookConfigurationMetadata": {
+      "type": "object",
+      "description": "The metadata of the playbook configuration resource.",
+      "properties": {
+        "createdDateTimeUtc": {
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time",
+          "description": "The playbook configuration creation date."
+        },
+        "createdBy": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The AAD object ID of the entity that created the playbook configuration."
+        },
+        "lastUpdatedDateTimeUtc": {
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time",
+          "description": "The playbook configuration last updated date."
+        },
+        "lastUpdatedBy": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The AAD object ID of the entity that last updated the playbook configuration."
+        }
+      }
+    },
+    "PlaybookConfigurationSource": {
+      "type": "object",
+      "description": "The source event types which evaluate the playbook configuration set of rules. For example - security alerts and recommended tasks.",
+      "properties": {
+        "eventSource": {
+          "type": "string",
+          "description": "A valid event source type."
+        },
+        "ruleSets": {
+          "type": "array",
+          "description": "A set of rules which evaluate upon event interception.",
+          "items": {
+            "$ref": "#/definitions/PlaybookConfigurationRuleSet"
+          }
+        }
+      }
+    },
+    "PlaybookConfigurationScope": {
+      "type": "object",
+      "description": "The subscription's resources scope",
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "The resources scope description."
+        },
+        "scopePath": {
+          "type": "string",
+          "description": "The resources scope path."
+        }
+      }
+    },
+    "PlaybookConfigurationAction": {
+      "type": "object",
+      "description": "The action that should be triggered.",
+      "properties": {
+        "resourceId": {
+          "type": "string",
+          "description": "The triggered resource id."
+        },
+        "resourceProviderNamespace": {
+          "type": "string",
+          "description": "The resource provider name."
+        },
+        "uri": {
```

Comment:
>uri [](start = 9, length = 3)

Are these going to be ARM calls or data plane calls? How will auth work here?

Reply:
There are many different types. Can be both potentially. Auth is handled differently on different action types.

The first action type supported, (logic app), the auth is send as part of the URI, as defined by LogicApp.

Reply:
Btw, if a property contains values that could be potentially considered a secret, you should not be returning it on GETs.

---
In reply to: [281746039](https://github.com/Azure/azure-rest-api-specs/pull/5548#discussion_r281746039) [](ancestors = 281746039)

Reply:
The secrets are not returned in the GETs (a text is returned specifying that value is a secret and can only be updated, not obtained)

Reply:
ok

---
In reply to: [282745636](https://github.com/Azure/azure-rest-api-specs/pull/5548#discussion_r282745636) [](ancestors = 282745636)